packages/react/src/hooks/useIsomorphicLayoutEffect.ts

```diff
@@ -0,0 +1,6 @@
+import { useEffect, useLayoutEffect } from 'react'
+import { isClient } from '../utils/isClient'
+
+const isClientEnv = isClient()
+
+export const useIsomorphicLayoutEffect = isClientEnv ? useLayoutEffect : useEffect
```

Comment:
How about simply now like below?

check window only
```suggestion
import { useEffect, useLayoutEffect } from 'react'

export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
```

or


check document only
```suggestion
import { useEffect, useLayoutEffect } from 'react'

export const useIsomorphicLayoutEffect = typeof document !== 'undefined' ? useLayoutEffect : useEffect
```

Reply:
What is different `isClient` function call case and `typeof document !== 'undefined'` case?
I don't know exactly about your intension.

Reply:
Then how about this. I thought we don't need to make extra code to make `isClient` as function yet.

1. no reusing `isClient` yet
2. function `isClient` will be contained in bundle
3. In our case, We don't have to recall it. so we don't have to make `isClient` as funciton

```suggestion
import { useEffect, useLayoutEffect } from 'react'
const isClient = typeof window !== 'undefined'
export const useIsomorphicLayoutEffect = isClient ? useLayoutEffect : useEffect
```

if we want to use as extra module, I think below is enough, not `isClient` as function

```ts
// src/utils/isClient.ts
export const isClient = typeof window !== 'undefined'
```

```suggestion
import { useEffect, useLayoutEffect } from 'react'
import { isClient } from '../utils' // using isClient as just boolean is okay

export const useIsomorphicLayoutEffect = isClient ? useLayoutEffect : useEffect
```